src/modules/paymentProcessor/abstracts/PP_CrossChainBase_v1.sol

```diff
@@ -0,0 +1,243 @@
+        public
+        view
+        virtual
+        returns (bytes memory bridgeData_)
+    {
+        return _paymentIdToBridgeData[paymentId_];
+    }
+
+    /// @inheritdoc IPP_CrossChainBase_v1
+    function getPaymentId() external view virtual returns (uint paymentId_) {
+        return _paymentId;
+    }
+
+    /// @inheritdoc IPaymentProcessor_v2
+    function unclaimable(
+        address client_,
+        address token_,
+        address paymentReceiver_
+    ) public view virtual override returns (uint amount_) {
+        return
+            _unclaimableAmountsForRecipient[client_][token_][paymentReceiver_];
+    }
+
+    //--------------------------------------------------------------------------
+    // External Functions
+
+    /// @inheritdoc IPaymentProcessor_v2
+    function claimPreviouslyUnclaimable(
+        address client_,
+        address token_,
+        address receiver_
+    ) external virtual override {
+        if (unclaimable(client_, token_, _msgSender()) == 0) {
+            revert Module__PaymentProcessor__NothingToClaim(
+                client_, _msgSender()
+            );
+        }
+
+        _claimPreviouslyUnclaimable(client_, token_, receiver_);
+    }
+
+    /// @inheritdoc IPaymentProcessor_v2
+    function cancelRunningPayments(IERC20PaymentClientBase_v2 client_)
+        external
+        virtual
+        onlyModule
+        validClient(address(client_))
+    {
+        // Implementation depends on specific bridge requirements
+        revert("Not implemented");
+    }
+
+    // -------------------------------------------------------------------------
+    // Internal Functions Implemented in Downstream Contract
+
+    /// @notice Execute the cross-chain bridge transfer.
+    /// @dev    Override this function to implement specific bridge logic.
+    /// @param  order_ The payment order containing all necessary transfer details.
+    function _executeBridgeTransfer(
+        IERC20PaymentClientBase_v2.PaymentOrder memory order_
+    ) internal virtual;
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    /// @notice used to claim the unclaimable amount of a particular `paymentReceiver` for a given payment client.
+    /// @param  client_ address of the payment client.
+    /// @param  token_ address of the payment token.
+    /// @param  paymentReceiver_ address of the paymentReceiver for which the unclaimable amount will be claimed.
+    function _claimPreviouslyUnclaimable(
+        address client_,
+        address token_,
+        address paymentReceiver_
+    ) internal virtual {
+        address sender = _msgSender();
+        uint amount = _unclaimableAmountsForRecipient[client_][token_][sender];
+        delete _unclaimableAmountsForRecipient[client_][token_][sender];
+
+        IERC20(token_).transfer(paymentReceiver_, amount);
+        emit TokensReleased(paymentReceiver_, address(token_), amount);
```

Comment:
may want to add the msg.sender in the event - it seems important to know who's tokens were claimed

Reply:
As this is a Payment Process event, the change here would effect quite a lot of code. I do agree however with your point. @FHieser any thoughts here?